src/AbstractSQLRules2SQL.ts

```diff
@@ -939,6 +940,17 @@ const typeRules: Dictionary<MatchFn> = {
 		const date = DateValue(getAbstractSqlQuery(args, 0), indent);
 		return `DATE(${date})`;
 	},
+	DateTrunc: (args, indent) => {
+		if (engine !== Engines.postgres) {
+			throw new Error(
+				'DATE_TRUNC function is only supported on postgres currently',
+			);
```

Comment:
I'd make this a `SyntaxError` and specify the engine it's not supported on in order to match the others
```suggestion
			throw new SyntaxError('DateTrunc is not supported on: ' + engine);
```